### PR TITLE
Document dimension of X/ZPowGate

### PR DESCRIPTION
**Pull request by 95-martin-orion**
As advertised. Everything except the "dimension" part is copied from the EigenGate constructor docstring.

Also opened #5647 for adding similar functionality + docs to YPowGate.

### CODE REVIEW

File: cirq-core/cirq/ops/common_gates.py [hunk 1]
```diff
@@ -92,6 +92,30 @@ class XPowGate(eigen_gate.EigenGate):
     def __init__(
         self, *, exponent: value.TParamVal = 1.0, global_shift: float = 0.0, dimension: int = 2
     ):
+        """Initialize an XPowGate.
+
+        Args:
+            exponent: The t in gate**t. Determines how much the eigenvalues of
+                the gate are scaled by. For example, eigenvectors phased by -1
```
Comment (viathor) [inline cirq-core/cirq/ops/common_gates.py:99]: s/scaled/phased/

The word "scaled" suggests a possible change to absolute value (which does not change - all eigenvalues of a unitary gate have unit absolute value).

(Moot if you choose to follow the suggestion below.)

File: cirq-core/cirq/ops/common_gates.py [hunk 2]
```diff
@@ -92,6 +92,30 @@ class XPowGate(eigen_gate.EigenGate):
...
+                eigenvectors unaffected by `gate**1`).
+            global_shift: Offsets the eigenvalues of the gate at exponent=1.
+                In effect, this controls a global phase factor on the gate's
+                unitary matrix. The factor is:
+
+                    exp(i * pi * global_shift * exponent)
+
+                For example, `cirq.X**t` uses a `global_shift` of 0 but
+                `cirq.rx(t)` uses a `global_shift` of -0.5, which is why
+                `cirq.unitary(cirq.rx(pi))` equals -iX instead of X.
```
Comment (viathor) [inline cirq-core/cirq/ops/common_gates.py:111]: Consider updating the formula on line 75 above to account for global shift. Then you can refer to whatever letter you choose to use for global shift in the formula from the description of the argument here. This is what you effectively do on line 98 for exponent=t.

The benefit of this is that it allows the reader to unambiguously identify the gate they will get for given arguments.

**Review by viathor — APPROVED** [review]

File: cirq-core/cirq/ops/common_gates.py (hunk 2 above)
Comment (95-martin-orion) [reply cirq-core/cirq/ops/common_gates.py:112]: Done, assuming I understood you correctly. PTAL

File: cirq-core/cirq/ops/common_gates.py (hunk 1 above)
Comment (95-martin-orion) [reply cirq-core/cirq/ops/common_gates.py]: Done, although this phrasing is copied from EigenGate. I'll update it there as well.

File: cirq-core/cirq/ops/common_gates.py (hunk 2 above)
Comment (viathor) [reply cirq-core/cirq/ops/common_gates.py:112]: Yes, that's exactly it! :-)

I haven't checked that the math is correct. Might be worth calling `np.linalg.eig(cirq.unitary(...))` on one of the gates with a few values of exponent and global_shift and check that the eigenvalues are correct.

**Review by viathor — APPROVED** [review]

File: cirq-core/cirq/ops/common_gates.py (hunk 2 above)
Comment (viathor) [reply cirq-core/cirq/ops/common_gates.py:112]: (To be clear: I think we do this in unit tests, but doing it manually would give you an opportunity for compare the output against the math in the comment.)

**Pull request merged**